src/aggregations/aggregateNewFields.js

```diff
@@ -28,7 +28,11 @@ module.exports = (ids, properties, limit) => {
 	]
 
 	properties.forEach((property) => {
-		aggregation[0].$match[property] = { $ne: null }
+		if (or) {
+			(aggregation[0].$match['$or'] = aggregation[0].$match['$or'] || []).push({ [property]: { $ne: null } })
```

Comment:
Smart one, but I had to read it a few times to understand what's going on 😃 Let's split it into two lines.

```js
const osProp = 
aggregation[0].$match['$or'] = aggregation[0].$match['$or'] || []
aggregation[0].$match['$or'].push({ [property]: { $ne: null } })
```

or (but I don't know if that's easier to read)

```js
aggregation[0].$match['$or'] = [...(aggregation[0].$match['$or'] || []), { [property]: { $ne: null } }]
```